Source/WTF/wtf/text/StringImpl.h

```diff
@@ -1369,19 +1369,19 @@ inline Ref<StringImpl> StringImpl::createByReplacingInCharacters(std::span<const
 }
 
 template<typename Func>
-inline Expected<std::invoke_result_t<Func, std::span<const char>>, UTF8ConversionError> StringImpl::tryGetUTF8(const Func& function, ConversionMode mode) const
+inline Expected<std::invoke_result_t<Func, std::span<const char8_t>>, UTF8ConversionError> StringImpl::tryGetUTF8(const Func& function, ConversionMode mode) const
 {
     if (is8Bit())
         return tryGetUTF8ForCharacters(function, span8());
     return tryGetUTF8ForCharacters(function, span16(), mode);
 }
 
 template<typename Func>
-inline Expected<std::invoke_result_t<Func, std::span<const char>>, UTF8ConversionError> StringImpl::tryGetUTF8ForCharacters(const Func& function, std::span<const LChar> characters)
+inline Expected<std::invoke_result_t<Func, std::span<const char8_t>>, UTF8ConversionError> StringImpl::tryGetUTF8ForCharacters(const Func& function, std::span<const LChar> characters)
 {
     if (characters.empty()) {
-        constexpr const char* emptyString = "";
-        return function(std::span(emptyString, emptyString));
+        constexpr std::array<char8_t, 1> empty { '\0' };
+        return function(std::span { empty.data(), empty.data() });
```

Comment:
This seems a bit peculiar. Why do we need a `span` with a zero byte past the end? I understand that's what we had before but it seems unlikely callers rely on data outside the `span`.

Reply:
Ah! I thought the same and an earlier version of my patch was returning a default constructed span. However, this introduced a lot of crashes. The issue seems to be the CString constructor from a span. It distinguishes the cases where the span is empty from the case where the same is empty and the pointer is null. It's trying to decide whether to constructor a null or an empty CString.

Reply:
Yea that sounds right to me: non-null pointer but length zero needed.

Reply:
Given the bot failures, it looks like the code may rely on the null terminator after all. I'll investigate when I have some spare cycles.